doc/jobs/ci_jobs.rst

```diff
@@ -0,0 +1,187 @@
+  which will perform maintenance tasks.
+
+  * The ``reconfigure-jobs`` job will (re-)configure the *CI* nightly and
+    overlay jobs on a regular basis (e.g. once every day).
+
+* **generate_ci_jobs.py** invokes *generate_ci_job.py* for both the nightly
+  configuration as well as the overlay configuration.
+* **generate_ci_job.py** generates a *CI* nightly job for for each platform
+  and architecture listed in the *CI build file*.
+* **generate_ci_script.py** generates a *shell* script which will run the
+  same tasks as the *CI* job on a local machine.
+
+The build process in detail
+---------------------------
+
+The actual build is performed within a Docker container in order to only make
+the declared dependencies available.
+
+The actual build process starts in the script *create_ci_task_generator.py*.
+It generates three Dockerfiles: one to perform the *create-workspace* task to
+populate the workspace and enumerate prerequisites, one to perform the
+*build-and-install* task, and one to perform the *build-and-test* task.
+
+A typical job runs 4 Dockerfiles in all.
+
+Create workspace
+^^^^^^^^^^^^^^^^
+
+This task is performed by the script *create_workspace.py*, the Colcon
+executable and the rosdep executable.
+
+The task performs the following steps:
+
+* Prepare the package sources
+
+  * Fetches the `.repos` file(s)
+  * Fetches the source repositories containing the package(s) to be built
+  * If necessary, switches the branch in those repositories
+
+* Select the packages which should be built, tested, and installed
+
+  * Marks the packages explicitly listed as ignored
+  * If necessary, ignores any packages not selected or in scope
+
+* Enumerate dependencies
+
+  * Lists the dependencies necessary to build and test the remaining ROS
+    packages in the workspace, as well as the dependencies of any ROS packages
+    in the underlay workspace(s) needed to support them.
+  * Uses the rosdep tool to resolve those dependency keys to package names
+
+Build and install
+^^^^^^^^^^^^^^^^^
+
+This task is identical to `the one used by the devel jobs <devel_jobs.rst#Build-and-install>`_.
+
+Build and test
+^^^^^^^^^^^^^^
+
+This task is identical to `the one used by the devel jobs <devel_jobs.rst#Build-and-test>`_.
+
+Known limitations
+^^^^^^^^^^^^^^^^^
+
+System dependency enumeration happens for all ROS packages that are part of the
+non-underlay workspace. This means that any non-ROS packages present in that
+workspace may need their dependencies explicitly called out for inclusion in
+the foundation package list, and also means that a missing dependency may be
+occluded by another package in the workspace correctly taking the same
+dependency.
+
+Run the *CI* job locally
+------------------------
+
+Example invocation
+^^^^^^^^^^^^^^^^^^
+
+The following commands run the *CI* job for the *ament_cmake_ros* package
+from ROS *Crystal* for Ubuntu *Bionic* *amd64*:
```

Comment:
What's the purpose of italicizing these words?

Reply:
Totally just following the convention used in the other job docs. I think maybe the italics denote things that are configurable?

Reply:
Thanks.